src/Components/Feature/ToDoListRq.tsx

```diff
@@ -0,0 +1,138 @@
+import { useQuery } from 'react-query';
+import { ChangeEvent, useState } from 'react';
+import { IToDoResp } from '../../Shared/types'
+import Todo from './ToDo'
+import FilterBar from './FIlterBar';
+
+const fetchTodos = async ({ pageNumber = 1}) => {
+    const queryParams = new URLSearchParams({
+        _page: pageNumber.toString(),
+        _limit: '2',
+        q: '',
+        _sort: '',
+    });
+
+    const response = await fetch(`http://localhost:5000/todos?${queryParams}`);
+    if (!response.ok) {
+        throw new Error('Network response was not ok');
+    }
+    return response.json();
+};
+
+
+
+const TodosListRqComponent = () => {
+    const [pageNumber, setPageNumber] = useState<any>(0)
+    const [originalTodos, setOriginalFilteredTodos] = useState<IToDoResp[]>([]);
+    const [filteredTodos, setFilteredTodos] = useState<IToDoResp[]>([]);
+    const [searchTerm, setSearchTerm] = useState<string>('');
+    const [sortBy, setSortBy] = useState<string>('');
+    const [statusFilter, setStatusFilter] = useState<string>('');
+
+    const onSuccess =  (data:IToDoResp[]) =>{
+        console.log(data)
+        setFilteredTodos(data);
+        setOriginalFilteredTodos(data)
+    }
+
+    const { isLoading, isError,error } = useQuery(['todos',pageNumber], ()=>fetchTodos(pageNumber),{
```

Comment:
here itself you can retrieve data instead of taking it from a success callback.
{ data,isLoading, isError, error }

Reply:
We can retrieve data from this source as well. However, for a more streamlined approach in managing data updates, it's beneficial to implement a separate function for reassignment with new data.